runtime/dart_init.cc

```diff
@@ -100,6 +100,7 @@ static const char* kDartCheckedModeArgs[] = {
     "--enable_type_checks",
     "--error_on_bad_type",
     "--error_on_bad_override",
+    "--assert_initializer",
```

Comment:
I believe it's the latter, and it makes sense to turn it on for all JIT mode configurations given that JIT doesn't always imply checked mode.  Moved it to kDartLanguageArgs.